pinot-broker/src/main/java/org/apache/pinot/broker/routing/builder/LowLevelRoutingTableBuilderUtil.java

```diff
@@ -32,59 +32,49 @@
 public class LowLevelRoutingTableBuilderUtil {
 
   /**
-   * Compute the map of allowed 'consuming' segments for each partition.
+   * Compute the map of allowed CONSUMING segments for each partition for routing purpose.
+   * <p>Within a partition, we only allow querying the first CONSUMING segment (segment with instances in CONSUMING
+   * state) for the following reasons:
+   * <ul>
+   *   <li>
+   *     If within a partition, there are multiple CONSUMING segments (typically caused by the delay of CONSUMING to
```

Comment:
Within a partition, there could be multiple CONSUMING segments (typically caused by the delay of CONSUMING to ONLINE state transition). We should only allow querying the first CONSUMING segment because it might contain records that overlapped with the records in the next segment (over-consumed).

Reply:
Done